modules/pam_limits/pam_limits.c

```diff
@@ -487,6 +487,46 @@ static int init_limits(pam_handle_t *pamh, struct pam_limit_s *pl, int ctrl)
     return retval;
 }
 
+/*
+ * Read the contents of /proc/sys/fs/<name>
+ * return 1 if conversion succeeds, result is in *valuep
+ * return 0 if conversion fails.
+ */
+static int
+value_from_proc_sys_fs(const char *name, rlim_t *valuep)
+{
+    char pathname[128];
+    char buf[128];
+    FILE *fp;
+    int retval;
+
+    retval = 0;
+
+    snprintf(pathname, sizeof(pathname), "/proc/sys/fs/%s", name);
+
+    if ((fp = fopen(pathname, "r")) != NULL) {
+	if (fgets(buf, sizeof(buf), fp) != NULL) {
+	    char *endptr;
+	    rlim_t value;
+
+#ifdef __USE_FILE_OFFSET64
+	    value = strtoull(buf, &endptr, 10);
+#else
+	    value = strtoul(buf, &endptr, 10);
+#endif
```

Comment:
Yes, configure.ac uses `AC_SYS_LARGEFILE` macro, but do we really need this very glibc-specific ifdef?
We could use `strtoull`unconditionally, assign the return value to a variable of type unsigned long long, and check for an overflow.  FWiW, we shall check the return value + `errno` anyway.  It's very unlikely given that `sysctl_nr_open` is less than `INT_MAX`, but a correct check is simple enough, e.g.:
```
char *endptr;
unsigned long long value;
errno = 0;
value = strtoull(buf, &endptr, 10);
if (endptr != buf &&
    (value != ULLONG_MAX || errno == 0) &&
    (unsigned long long) (rlim_t) value == value) {
        *valuep = (rlim_t) value;
        retval = 1;
}
```

Reply:
The same construct is used inside `process_limit()`, so I stuck with that.
But, again: no problem here.

Reply:
Thanks.
Looks like the module would benefit from some cleaning up, but that's out of scope of this PR.